### ISSUE TITLE
Improvements to torch.compile integration

### ISSUE THREAD

**Issue by msaroufim** [pytorch/serve#2135]
### 🚀 The feature

Last year we introduced `torch.compile()` support but it was done fairly quickly and there's a few more improvements that make sense to make

- [ ] `check_pt2_enabled()` is a bad name and should instead be `check_pt2_available()`
- [ ] We can gate compilation and inferences with a compiled model with a property in model.config, this makes it easier to rollback to an uncompiled model if there are issues in prod
- [ ] Enabling tensor cores should be opt in since it has an accuracy impact or at least should have a warning
- [ ] Not all compiler passes and modes are available from `compile.json` but should be
- [ ] Rework logger.warning to .info when applicable
- [ ] Integrate `torch._dynamo.export()`
- [ ] Update intall instructions to --pre torch, torch[dynamo] is shown as not available now
- [ ] Update docs to show that holding both regular and compiled model in memory doesnt introduce much overhead
- [ ] Describe issues that model load will be slow and that recompilations will happen frequently - so for example if torch.compile is used we should recommend users make their batch sizes small and their batch delay large so the batch is always filled out

### Motivation, pitch

n

**Comment by msaroufim** [comment]
Just ran a quick experiment with this file, my goal was to confirm that holding both the model and optimized model in memory introduces no memory overhead so we can easily switch from one to another. It works just fine! I always see about  the same memory utilization

```python
import torch
torch.set_float32_matmul_precision('high')

class ToyModel(torch.nn.Module):
    def __init__(self):
        super().__init__()

        self.linear = torch.nn.Linear(1-000-000, 1-000-000)
        self.linear1 = torch.nn.Linear(1-000-000, 1-000-000)
        self.linear2 = torch.nn.Linear(1-000-000, 1-000-000)
        self.linear3 = torch.nn.Linear(1-000-000, 1-000-000)
        self.relu = torch.nn.ReLU()

    
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        return self.relu(self.linear3(self.linear2(self.linear1(self.linear(x)))))
    

m = ToyModel().cuda()
opt_m = torch.compile(m)
for _ in range(10000000):
    out = m(torch.ones(1-000-000, 1-000-000).cuda())
    out_opt = opt_m(torch.ones(1-000-000, 1-000-000).cuda())
```
![Screenshot 2023-03-02 at 10 08 17 AM](https://user-images.githubusercontent.com/3282513/222514445-3ce3170a-4a79-4892-a16d-d317044abc99.png)
